crates/sui-framework/sources/governance/validator.move

```diff
@@ -584,18 +620,39 @@ module sui::validator {
         validate_metadata(&self.metadata);
     }
 
+    /// Update protocol public key of this validator
+    public(friend) fun update_protocol_pubkey(self: &mut Validator, protocol_pubkey: vector<u8>, proof_of_possession: vector<u8>) {
+        // TODO move proof of possession verification to the native function
+        verify_proof_of_possession(proof_of_possession, self.metadata.sui_address, protocol_pubkey);
+        self.metadata.protocol_pubkey_bytes = protocol_pubkey;
+        self.metadata.proof_of_possession = proof_of_possession;
+        validate_metadata(&self.metadata);
+    }
+
     /// Update network public key of this validator, taking effects from next epoch
     public(friend) fun update_next_epoch_network_pubkey(self: &mut Validator, network_pubkey: vector<u8>) {
         self.metadata.next_epoch_network_pubkey_bytes = option::some(network_pubkey);
         validate_metadata(&self.metadata);
     }
 
+    /// Update network public key of this validator
+    public(friend) fun update_network_pubkey(self: &mut Validator, network_pubkey: vector<u8>) {
+        self.metadata.network_pubkey_bytes = network_pubkey;
+        validate_metadata(&self.metadata);
+    }
+
     /// Update Narwhal worker public key of this validator, taking effects from next epoch
     public(friend) fun update_next_epoch_worker_pubkey(self: &mut Validator, worker_pubkey: vector<u8>) {
         self.metadata.next_epoch_worker_pubkey_bytes = option::some(worker_pubkey);
         validate_metadata(&self.metadata);
     }
 
+    /// Update Narwhal worker public key of this validator
+    public(friend) fun update_worker_pubkey(self: &mut Validator, worker_pubkey: vector<u8>) {
```

Comment:
We should add asserts in every of these functions to verify the validator is a candidate. Also renaming it as `update_candiate_worker_pubkey` could make its intention more clear.